nanofi/include/api/nanofi.h

```diff
@@ -165,6 +355,31 @@ uint8_t remove_attribute(flow_file_record*, char *key);
 
 int transmit_flowfile(flow_file_record *, nifi_instance *);
 
+/**
+ * Adds a custom processor for later instantiation
+ * @param name name of the processor
+ * @param logic the callback to be invoked when the processor is triggered
+ * @return 0 on success, -1 otherwise (name already in use for eg.)
+ **/
+int add_custom_processor(const char * name, processor_logic* logic);
```

Comment:
processor_logic doesn't encapsulate the potential of "scheduling" May want to keep that fundamentally so that we allow configuration items to be loaded only once. The Python code left that as a todo

Reply:
Typically the session can define a mapping of properties and can call a "scheduling" piece before every iteration. In some cases, this could be at the outset of execution of a custom processor. Effectively before their actions are invoked, the processor should be configured. We can make that stipulation a priori unless properties are dynamic, in which case we've sort of tabled that for the interim. MINIFICPP-694 is a good follow on to that.